README.md

```diff
@@ -1,38 +1,98 @@
-* the next key part defines the deployment service, e.g. "sbdb" or "nbdb"
-* after that, we have a database server name, e.g. `OVN_Northbound`, `OVN_Southbound`, `_Server`. There  is an 
-  additional internal entry `_` for locks and comments.
-* next key element defines table names according to the relevant schema. Locks and comments are stored in the internal 
-  database under the `_locks` and `_comments` entries.
-* the last key element is the `uuid` of a table row, `lock ID`, or a comment's `timestamp`.
-* in order to guarantee that `_Server/Database` will contains only entries per running databases, its last key element
-  is not uuid, but the database server name.
-  
-The above explanation can be demonstrated as:
-- data:           <prefix>/<service>/<dbName>/<table>/<uuid> --> <row>
-- locks:          <prefix>/<service>/_/_locks/<lockid> --> nil
-- comments:       <prefix>/<service>/_/_comments/<timestamp> --> <comment>
+## Implementation of ovsdb-etcd
+
+The objects are mapped to etcd as follows:
+
+```
+data:           <prefix>/<service>/<dbName>/<table>/<uuid> --> <row>
+locks:          <prefix>/<service>/_/_locks/<lockid> --> nil
+comments:       <prefix>/<service>/_/_comments/<timestamp> --> <comment>
+```
+
+As ovsdb transactions have different semantics than etcd transactions, the
+initial implementation consists of:
+- receiving the transaction-request in ovsdb semantics
+- catching the global lock
+- etcd get the pre-transaction **tables**
+- process ovsdb semantics on memory
+- etcd post the post-transaction **rows**
+- releasing the global lock
+- sending the transaction-response in ovsdb semantics
+
+An ovsdb monitor enrollment is implemented as:
+- receiving monitor-request in ovsdb semantics
+- enrolling to get etcd events per **tables**
+- sending monitor-response in ovsdb semantics
+
+An ovsdb monitor event is implemented as:
+- receiving monitor-event in etcd semantics
+- translating the etcd event to ovsdb event
+- sending monitor-event is ovsdb semantics
+
+## To start developing ovsdb-etcd
+
+You must have a working [Go Environment](https://golang.org/doc/install).
+
+Then clone ovn-kubernetes from our fork:
+
+```bash
+git clone https://github.com/ibm/ovn-kubernetes
+pushd ovn-kubernetes
+git checkout ovsdb-etcd2
+export OVN_KUBERNETES_ROOT=$PWD
+popd
+```
+
+Then clone ovsdb-etcd and build and push docker images:
+
+```bash
+git clone https://github.com/IBM/ovsdb-etcd
+pushd ovsdb-etcd
+make docker-build
+popd
+```
+
+Setup your docker.io credentials:
+
+```
+export CR_PASSWORD=<DOCKER_PASSWORD>
+export CR_USERNAME=<DOCKER_USENAME>
+export OVSDB_ETCD_REPOSITORY=$CR_USERNAME
+```
+Login to docker and push (optional):
+
+```
+echo $CR_PAT | docker login $CR_REPO -u $CR_USERNAME --password-stdin
+make docker-push
+```
+
+Disable tcpdump logging (optional):
+
+```
+export OVNDB_ETCD_TCPDUMP='false'
+```
+
+Deploy onto a kind cluster:
+
+
+```bash
+pushd ovsdb-etcd
```

Comment:
why do we need all thes pusgd ovsdb-etcd/popd ? if user works with ovsdb-etcd so he is cd to the ovsdb-etcd and works from its root dir

Reply:
removed